airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py

```diff
@@ -165,143 +158,34 @@ def get_time_ordered_file_infos(self) -> List[FileInfo]:
         """
         return sorted(self.pattern_matched_filepath_iterator(self.filepath_iterator()), key=lambda file_info: file_info.last_modified)
 
-    def _get_schema_map(self) -> Mapping[str, Any]:
-        return_schema: Dict[str, Any] = None
-        if self._schema != {}:
-            return_schema = deepcopy(self._schema)
-        else:  # we have no provided schema or schema state from a previous incremental run
-            return_schema = self._get_master_schema()
+    @property
+    def _raw_schema(self) -> Mapping[str, Any]:
+        if self._user_input_schema != {}:
+            return deepcopy(self._user_input_schema)
```

Comment:
nit: would it be safer to do it like this:
```suggestion
        if self._user_input_schema and isinstance(self._user_input_schema, dict):
            return deepcopy(self._user_input_schema)
```
I don't really get the requirement for a deepcopy here: we'll create a copy on each property access and I'm not sure _raw_schema is mutated after access. Is it?